### PR TITLE
feat: add metrics (part 3)

### PR DESCRIPTION
**Pull request by arithmetic1728**
This PR adds `x-goog-api-client` header to
- access token and id token refresh requests, for compute engine credentials / user credentials / service account credentials / impersonated credentials
- reauth start and continue requests
- metadata server ping requests

Previous PRs:
Part 1: #1298 
Part 2: #1303

### CODE REVIEW

**Comment by arithmetic1728** [comment]
@aeitzman please take a look at the 3PI unit test changes. 3PI credentials use impersonated credentials, since in this PR I added x-goog-api-client header to impersonated credential's refresh method, I made the unit test changes accordingly for this new header.

File: google/auth/compute_engine/_metadata.py [hunk 1]
```diff
@@ -121,13 +122,13 @@ def ping(request, timeout=_METADATA_DEFAULT_TIMEOUT, retry_count=3):
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
     retries = 0
+    headers = _METADATA_HEADERS.copy()
```
Comment (clundin25) [inline google/auth/compute_engine/_metadata.py:125]: Nice catch on this bug :)

**Review by clundin25 — APPROVED** [review]

**Comment by aeitzman** [comment]
> @aeitzman please take a look at the 3PI unit test changes. 3PI credentials use impersonated credentials, since in this PR I added x-goog-api-client header to impersonated credential's refresh method, I made the unit test changes accordingly for this new header.

ack - will take a look today

File: tests/test_external_account.py [hunk 2]
```diff
@@ -26,6 +26,8 @@
 from google.auth import transport
 
 
+METRICS_HEADER_VALUE = "gl-python/3.7 auth/1.1 auth-request-type/at cred-type/imp"
```
Comment (aeitzman) [inline tests/test_external_account.py:29]: nit: Can this be named IMPERSONATE_ACCESS_TOKEN_REQUEST_METRICS_HEADER_VALUE to be consistent with the AWS tests and also to make it more clear this is only expected for SA impersonation requests?

**Review by aeitzman — APPROVED** [review]
One small nit, everything else LGTM

File: tests/test_external_account.py (hunk 2 above)
Comment (arithmetic1728) [reply tests/test_external_account.py:29]: updated the name

**Pull request merged**